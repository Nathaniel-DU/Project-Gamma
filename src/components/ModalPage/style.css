.modal-btn {
    background-color: #ADEFD1FF!important;
    width: 50%;
    border: black 1px solid;
    padding: 10px;
    border-radius: 15px;
    margin: auto;
    display: block;
    color: black;
}

.dismiss-modal {
    background-color: #ADEFD1FF!important;
    padding: 10px;
    border-radius: 15px;
    
}

.modal-header {
    color: #00203FFF;
    
}

.fa-icon {
    margin-right: 15px;
}