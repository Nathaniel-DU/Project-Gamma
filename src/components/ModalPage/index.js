import React, { Component } from 'react';
import { MDBContainer, MDBBtn, MDBModal, MDBModalHeader, MDBModalFooter } from 'mdbreact';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStopCircle, faMapMarkedAlt, faPhoneVolume, faCar, faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
import './style.css';
import axios from 'axios';

class ModalPage extends Component {
  
  state = {
      modal1: false,
      modal2: false,
      modal3: false,
      modal4: false
  }

  toggle = nr => () => {
      let modalNumber = `modal` + nr;
      switch(modalNumber){
      case `modal1`:
          this.getMyLocation();
          axios.get(`/event/excuse`)
              .catch(err => console.log(err));
          break;
      case `modal2`: 
          this.getMyLocation();
          axios.get(`/event/ride`)
              .catch(err => console.log(err));
          break;
      case `modal3`:
          this.getMyLocation();
          axios.get(`/event/emergency`)
              .catch(err => console.log(err));
          break;
      case `modal4`: 
          this.getMyLocation();
          break;
      default:
          break;
      }
      this.setState({
          [modalNumber]: !this.state[modalNumber]
      });
  }

  toggleOff = nr => () => {
      let modalNumber = `modal` + nr;
      this.setState({
          [modalNumber]: !this.state[modalNumber]
      });
  }

  constructor() {
      super();

      this.state = {
          latitude: ``,
          longitude: ``,
          eventStarted: false
      };

      this.getMyLocation = this.getMyLocation.bind(this);
      this.startTrip = this.startTrip.bind(this);
      this.stopTrip = this.stopTrip.bind(this);
  }

  getMyLocation() {
    
      const location = window.navigator && window.navigator.geolocation;

      if (location) {
          location.getCurrentPosition((position) => {
              this.setState({
                  latitude: position.coords.latitude,
                  longitude: position.coords.longitude,
              });
              
              axios.put(`/event/updatelocation`, {
                  lat: this.state.latitude,
                  long: this.state.longitude,
              })
                  .catch(err => console.log(err));
          }, () => {
              this.setState({ latitude: `err-latitude`, longitude: `err-longitude` });
          });
      
      }
  }
 

  startTrip() {
      this.setState({ eventStarted: true });
  }

  stopTrip(){
      axios.put(`/event/stop`);
      window.location.reload();
  }

  render() {
      return (
          <MDBContainer>
              <MDBBtn className="modal-btn mint-button" color="primary" onClick={this.stopTrip}><FontAwesomeIcon className="fa-icon" icon={faStopCircle} />Stop Trip</MDBBtn>
              <MDBBtn className="modal-btn mint-button" color="primary" onClick={this.toggle(1)}><FontAwesomeIcon className="fa-icon" icon={faPhoneVolume} />Excuse Call</MDBBtn>
              <MDBModal isOpen={this.state.modal1} toggle={this.toggleOff(1)} centered>
                  <MDBModalHeader className="modal-header" toggle={this.toggleOff(1)}>Message Sent</MDBModalHeader>
                  <MDBModalFooter>
                      <MDBBtn className="dismiss-modal" onClick={this.toggleOff(1)}>Dismiss</MDBBtn>
                  </MDBModalFooter>
              </MDBModal>
              <MDBBtn className="modal-btn mint-button" color="primary" onClick={this.toggle(2)}><FontAwesomeIcon className="fa-icon" icon={faCar} />Request Pickup</MDBBtn>
              <MDBModal isOpen={this.state.modal2} toggle={this.toggleOff(2)} centered>
                  <MDBModalHeader className="modal-header" toggle={this.toggleOff(2)}>Message sent</MDBModalHeader>
                  <MDBModalFooter>
                      <MDBBtn className="dismiss-modal" onClick={this.toggleOff(2)}>Dismiss</MDBBtn>
                  </MDBModalFooter>
              </MDBModal>
              <MDBBtn className="modal-btn mint-button" color="primary" onClick={this.toggle(3)}><FontAwesomeIcon className="fa-icon" icon={faExclamationCircle} />Request Emergency Assistance</MDBBtn>
              <MDBModal isOpen={this.state.modal3} toggle={this.toggleOff(3)} centered>
                  <MDBModalHeader className="modal-header" toggle={this.toggleOff(3)}>Message Sent</MDBModalHeader>
                  <MDBModalFooter>
                      <MDBBtn className="dismiss-modal" onClick={this.toggleOff(3)}>Dismiss</MDBBtn>
                  </MDBModalFooter>
              </MDBModal>
              <MDBBtn className="modal-btn mint-button" color="primary" onClick={this.toggle(4)}><FontAwesomeIcon className="fa-icon" icon={faMapMarkedAlt} />Update Location</MDBBtn>
              <MDBModal isOpen={this.state.modal4} toggle={this.toggleOff(4)} centered>
                  <MDBModalHeader className="modal-header" toggle={this.toggleOff(4)}>Location Updated</MDBModalHeader>
                  <MDBModalFooter>
                      <MDBBtn className="dismiss-modal" onClick={this.toggleOff(4)}>Dismiss</MDBBtn>
                  </MDBModalFooter>
              </MDBModal>
          </MDBContainer>
      
      );
  }
}

export default ModalPage;
