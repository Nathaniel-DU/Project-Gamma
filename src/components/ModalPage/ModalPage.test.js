import React from 'react';
import { shallow } from 'enzyme';
import ModalPage from './index';

describe(`ModalPage`, () => {
    it(`should render correctly in "debug" mode`, () => {
        const component = shallow(<ModalPage debug />);
        expect(component).toMatchSnapshot();
    });
});