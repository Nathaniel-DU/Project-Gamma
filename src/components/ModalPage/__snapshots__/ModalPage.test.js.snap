// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPage should render correctly in "debug" mode 1`] = `
<be
  fluid={false}
  tag="div"
>
  <qe
    className="modal-btn mint-button"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f28d",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm96 328c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h160c8.8 0 16 7.2 16 16v160z",
          ],
          "iconName": "stop-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Stop Trip
  </qe>
  <qe
    className="modal-btn mint-button"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            384,
            512,
            Array [],
            "f2a0",
            "M97.333 506.966c-129.874-129.874-129.681-340.252 0-469.933 5.698-5.698 14.527-6.632 21.263-2.422l64.817 40.513a17.187 17.187 0 0 1 6.849 20.958l-32.408 81.021a17.188 17.188 0 0 1-17.669 10.719l-55.81-5.58c-21.051 58.261-20.612 122.471 0 179.515l55.811-5.581a17.188 17.188 0 0 1 17.669 10.719l32.408 81.022a17.188 17.188 0 0 1-6.849 20.958l-64.817 40.513a17.19 17.19 0 0 1-21.264-2.422zM247.126 95.473c11.832 20.047 11.832 45.008 0 65.055-3.95 6.693-13.108 7.959-18.718 2.581l-5.975-5.726c-3.911-3.748-4.793-9.622-2.261-14.41a32.063 32.063 0 0 0 0-29.945c-2.533-4.788-1.65-10.662 2.261-14.41l5.975-5.726c5.61-5.378 14.768-4.112 18.718 2.581zm91.787-91.187c60.14 71.604 60.092 175.882 0 247.428-4.474 5.327-12.53 5.746-17.552.933l-5.798-5.557c-4.56-4.371-4.977-11.529-.93-16.379 49.687-59.538 49.646-145.933 0-205.422-4.047-4.85-3.631-12.008.93-16.379l5.798-5.557c5.022-4.813 13.078-4.394 17.552.933zm-45.972 44.941c36.05 46.322 36.108 111.149 0 157.546-4.39 5.641-12.697 6.251-17.856 1.304l-5.818-5.579c-4.4-4.219-4.998-11.095-1.285-15.931 26.536-34.564 26.534-82.572 0-117.134-3.713-4.836-3.115-11.711 1.285-15.931l5.818-5.579c5.159-4.947 13.466-4.337 17.856 1.304z",
          ],
          "iconName": "phone-volume",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Excuse Call
  </qe>
  <o
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    centered={true}
    disableBackdrop={false}
    disableFocusTrap={true}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalStylesWithoutBackdrop={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    modalTransitionTimeout={300}
    noClickableBodyWithoutBackdrop={false}
    overflowScroll={true}
    position=""
    role="dialog"
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <tt
      className="modal-header"
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    >
      Message Sent
    </tt>
    <et>
      <qe
        className="dismiss-modal"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Dismiss
      </qe>
    </et>
  </o>
  <qe
    className="modal-btn mint-button"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f1b9",
            "M499.99 176h-59.87l-16.64-41.6C406.38 91.63 365.57 64 319.5 64h-127c-46.06 0-86.88 27.63-103.99 70.4L71.87 176H12.01C4.2 176-1.53 183.34.37 190.91l6 24C7.7 220.25 12.5 224 18.01 224h20.07C24.65 235.73 16 252.78 16 272v48c0 16.12 6.16 30.67 16 41.93V416c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h256v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-54.07c9.84-11.25 16-25.8 16-41.93v-48c0-19.22-8.65-36.27-22.07-48H494c5.51 0 10.31-3.75 11.64-9.09l6-24c1.89-7.57-3.84-14.91-11.65-14.91zm-352.06-17.83c7.29-18.22 24.94-30.17 44.57-30.17h127c19.63 0 37.28 11.95 44.57 30.17L384 208H128l19.93-49.83zM96 319.8c-19.2 0-32-12.76-32-31.9S76.8 256 96 256s48 28.71 48 47.85-28.8 15.95-48 15.95zm320 0c-19.2 0-48 3.19-48-15.95S396.8 256 416 256s32 12.76 32 31.9-12.8 31.9-32 31.9z",
          ],
          "iconName": "car",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Request Pickup
  </qe>
  <o
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    centered={true}
    disableBackdrop={false}
    disableFocusTrap={true}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalStylesWithoutBackdrop={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    modalTransitionTimeout={300}
    noClickableBodyWithoutBackdrop={false}
    overflowScroll={true}
    position=""
    role="dialog"
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <tt
      className="modal-header"
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    >
      Message sent
    </tt>
    <et>
      <qe
        className="dismiss-modal"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Dismiss
      </qe>
    </et>
  </o>
  <qe
    className="modal-btn mint-button"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Request Emergency Assistance
  </qe>
  <o
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    centered={true}
    disableBackdrop={false}
    disableFocusTrap={true}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalStylesWithoutBackdrop={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    modalTransitionTimeout={300}
    noClickableBodyWithoutBackdrop={false}
    overflowScroll={true}
    position=""
    role="dialog"
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <tt
      className="modal-header"
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    >
      Message Sent
    </tt>
    <et>
      <qe
        className="dismiss-modal"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Dismiss
      </qe>
    </et>
  </o>
  <qe
    className="modal-btn mint-button"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f5a0",
            "M288 0c-69.59 0-126 56.41-126 126 0 56.26 82.35 158.8 113.9 196.02 6.39 7.54 17.82 7.54 24.2 0C331.65 284.8 414 182.26 414 126 414 56.41 357.59 0 288 0zm0 168c-23.2 0-42-18.8-42-42s18.8-42 42-42 42 18.8 42 42-18.8 42-42 42zM20.12 215.95A32.006 32.006 0 0 0 0 245.66v250.32c0 11.32 11.43 19.06 21.94 14.86L160 448V214.92c-8.84-15.98-16.07-31.54-21.25-46.42L20.12 215.95zM288 359.67c-14.07 0-27.38-6.18-36.51-16.96-19.66-23.2-40.57-49.62-59.49-76.72v182l192 64V266c-18.92 27.09-39.82 53.52-59.49 76.72-9.13 10.77-22.44 16.95-36.51 16.95zm266.06-198.51L416 224v288l139.88-55.95A31.996 31.996 0 0 0 576 426.34V176.02c0-11.32-11.43-19.06-21.94-14.86z",
          ],
          "iconName": "map-marked-alt",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Update Location
  </qe>
  <o
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    centered={true}
    disableBackdrop={false}
    disableFocusTrap={true}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalStylesWithoutBackdrop={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    modalTransitionTimeout={300}
    noClickableBodyWithoutBackdrop={false}
    overflowScroll={true}
    position=""
    role="dialog"
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <tt
      className="modal-header"
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    >
      Location Updated
    </tt>
    <et>
      <qe
        className="dismiss-modal"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Dismiss
      </qe>
    </et>
  </o>
</be>
`;
